plugins/coatils.py

```diff
@@ -124,3 +126,112 @@ def ls(self, msg, match):
+    def construct_settings(settings):
+        settings = settings.strip().split()
+
+        def is_setting(x): return '=' in x  # Ignore PycodestyleBear (E731)
+        setting_dict = OrderedDict()
+        for candidate in settings:
+            if not is_setting(candidate):  # bear
+                setting_dict[candidate.strip()] = dict()
+            else:
+                key, value = candidate.strip().split('=')
+                setting_dict[list(setting_dict.keys())[-1]][key] = value
+        return setting_dict
+
+    @staticmethod
+    def position(stl, stc, enl, enc):
+        if stc is None and enc is None:
+            if stl != enl:
+                return 'Between lines {} and {}'.format(stl, enl)
+            else:
+                return 'At line {}'.format(stl)
+        if stc is None and enc is not None:
+            return 'Between line {} and position {}:{}'.format(stl, enl, enc)
+        if stc is not None and enc is None:
+            return 'Between position {}:{} and line {}'.format(stl, stc, enl)
+        if stl == enl:
+            if stc == enc:
+                return "At {}:{}".format(stl, stc)
+            else:
+                return "At line {}, between col {} and {}".format(stl, stc, enc)
+        else:
+            return "Between positions {}:{} and {}:{}".format(stl, stc,
+                                                              enl, enc)
+
+    # Ignore PycodestyleBear, LineLengthBear
+    @re_botcmd(pattern=r'run\s+(\w+)((?:\s+\w+(?:\s+\w+=\w+)*)+)\n+```\n([\s\S]+)\n```')
+    def run(self, msg, match):
+        """
+        Run coala over the given code.
+
+        Example: `run Bear1 setting1=something setting2=something Bear2\ncode`
+        """
+        lang = match.group(1)
+        bear_settings = Coatils.construct_settings(match.group(2))
+        code = match.group(3)
+
+        yield 'coala analysis in progress...'
+
+        data = {
+            "sections": {
+                "corobo": {
+                    "files": "**.gyp",
+                    "bears": dict(bear_settings)
+                }
+            },
+            "mode": "coala",
+            "language": lang,
+            "file_data": code
+        }
+
+        # Ignore InvalidLinkBear, this only accepts post requests
+        rq = requests.post('https://api.gitmate.io/coala_online/', json=data)
+        try:
+            results = rq.json()['response']['results']['corobo']
+        except json.JSONDecodeError:
+            went_wrong = '\n - '.join([
+                'Is the bear name correct? Note that bear names are '
+                'case sensitive.'
+                'Are all required settings provided? If a required setting is '
+                'not provided, analysis will fail.'
+            ])
+            yield 'Something went wrong, things to check for:\n' + went_wrong
+            self.log.exception('Something went wrong, please try again')
+        else:
+            yield ('Here is what I think is wrong: '
```

Comment:
merging these two messages into one logical line is not helpful.  One of these messages is actually a part of the messages that follows.  Make the exit condition more clear.

```
if not results:
    yield ....
    return

yield ...
for result in results:
    ...
```

However personally I think you should build the result string and do a single yield.  Yielding individual lines isnt going to have much benefit as this `result` loop is tightly bound since everything is in memory - yielding chunks is only going to slow it down with IO.